x-pack/plugins/fleet/server/services/epm/elasticsearch/ilm/install.ts

```diff
@@ -57,7 +58,8 @@ export async function installILMPolicy(
           { logger }
         );
       } catch (err) {
-        throw new Error(err.message);
+        logger.warn(`Couldn't install ilm policies: ${err.message}`);
+        throw new FleetError(`Couldn't install ilm policies: ${err.message}`);
```

Comment:
`FleetError`s result in 400s, right?

Reply:
From what I could see `FleetError` is checked in [this function](https://github.com/elastic/kibana/blob/21fc4cc3079db00d9898ec510049df03309b3faa/x-pack/plugins/fleet/server/errors/handlers.ts#L93-L124) and if it's not one of the errors explicitly defined still returns a 500. 

However I decided to update most of them under `epm` to highlight all the cases, we might catch something we didn't notice 😅 Plus, I'm FLeetError comes with the logger (so I can remove the duplicate logger lines I added)

Reply:
Based on the previous comments I'll change this one to `PackageInvalidArchiveError` that is a 400.

Reply:
> From what I could see `FleetError` is checked in [this function](https://github.com/elastic/kibana/blob/21fc4cc3079db00d9898ec510049df03309b3faa/x-pack/plugins/fleet/server/errors/handlers.ts#L93-L124) and if it's not one of the errors explicitly defined still returns a 500.

But it will enter on this `if`, call `getHTTPResponseCode` and result in a 400 code, right? :thinking: https://github.com/elastic/kibana/blob/21fc4cc3079db00d9898ec510049df03309b3faa/x-pack/plugins/fleet/server/errors/handlers.ts#L96